src/inputs/internal/ComboBoxBase.tsx

```diff
@@ -286,65 +253,34 @@ export function ComboBoxBase<O, V extends Value>(props: ComboBoxBaseProps<O, V>)
     },
   });
 
+  const selectedKeys = useMemo(() => {
+    return selectedOptions.map((o) => valueToKey(getOptionValue(o)));
+  }, [selectedOptions, getOptionValue]);
   // @ts-ignore - `selectionManager.state` exists, but not according to the types
   state.selectionManager.state = useMultipleSelectionState({
     selectionMode: multiselect ? "multiple" : "single",
     // Do not allow an empty selection if single select mode
     disallowEmptySelection: !multiselect,
-    selectedKeys: fieldState.selectedKeys,
+    selectedKeys,
     onSelectionChange,
   });
 
-  // Ensure we reset if the field's values change and the user is not actively selecting options.
-  useEffect(
-    () => {
-      if (!state.isOpen && !areArraysEqual(values, fieldState.selectedKeys)) {
-        setFieldState((prevState) => {
-          const selectedOptions = prevState.allOptions.filter((o) => values?.includes(getOptionValue(o)));
-          return {
-            ...prevState,
-            selectedKeys: selectedOptions?.map((o) => valueToKey(getOptionValue(o))) ?? [],
-            inputValue:
-              selectedOptions.length === 1
-                ? getOptionLabel(selectedOptions[0])
-                : multiselect && selectedOptions.length === 0
-                ? nothingSelectedText
-                : "",
-            selectedOptions: selectedOptions,
-          };
-        });
-      }
-    },
-    // TODO: validate this eslint-disable. It was automatically ignored as part of https://app.shortcut.com/homebound-team/story/40033/enable-react-hooks-exhaustive-deps-for-react-projects
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [values],
-  );
-
-  // Re-sync fieldState.allOptions
-  useEffect(
-    () => {
-      setFieldState((prevState) => {
-        const selectedOptions = options.filter((o) => values?.includes(getOptionValue(o)));
-        return {
-          ...prevState,
-          selectedKeys: selectedOptions?.map((o) => valueToKey(getOptionValue(o))) ?? [],
-          inputValue:
-            selectedOptions.length === 1
-              ? getOptionLabel(selectedOptions[0])
-              : multiselect && selectedOptions.length === 0
-              ? nothingSelectedText
-              : "",
-          selectedOptions: selectedOptions,
-          filteredOptions: options,
-          allOptions: options,
-        };
-      });
-    },
-    // We're primarily only re-setting `allOptions`, and so recalc selected as well, but we don't
-    // want to depend on values/etc., b/c we'll defer to their useEffects to update their state
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [options],
-  );
+  // Reset inputValue when closed or selected changes
```

Comment:
Kinda surprised, this useEffect ends up being a pretty cute place to handle resetting inputValue as various conditions change.